nncf/openvino/quantization/quantize_model_if_op.py

```diff
@@ -0,0 +1,226 @@
+            then_dataset.append(data_item)
+        else:
+            for name in else_model_input_names:
+                data_item.append(results[name])
+            else_dataset.append(data_item)
+    nncf_logger.info(f"The length of dataset for then body is {len(then_dataset)}")
+    nncf_logger.info(f"The length of dataset for else body is {len(else_dataset)}")
+    return Dataset(then_dataset), Dataset(else_dataset)
+
+
+def _extract_if_submodel(
+    model_transformer: ModelTransformer, if_node: NNCFNode, if_submodel_condition: bool
+) -> ov.Model:
+    """
+    Returns if submodel of If node laying on an input port if_submodel_port_id of If node.
+
+    :param model_transformer: ModelTransformer instance.
+    :param if_node: If node.
+    :param if_submodel_condition: If True returns True submodel of If node, otherwise - False submodel.
+    :return: If submodel.
+    """
+    transformation_layout = TransformationLayout()
+    command = OVBackend.create_extract_if_subgraph_command(if_node, if_submodel_condition)
+    transformation_layout.register(command)
+    return model_transformer.transform(transformation_layout)
+
+
+def _update_if_submodel(
+    model_transformer: ModelTransformer, if_node: NNCFNode, if_submodel_condition: bool, submodel: ov.Model
+) -> ov.Model:
+    """
+    Update submodel of If node.
+
+    :param model_transformer: ModelTransformer instance.
+    :param if_node: If node.
+    :param if_submodel_condition: Condition of If node submodel.
+    :param submodel: New submodel.
+    :return: Updated model with a new submodel.
+    """
+    transformation_layout = TransformationLayout()
+    command = OVBackend.create_update_subgraph_command(if_node, if_submodel_condition, submodel)
+    transformation_layout.register(command)
+    return model_transformer.transform(transformation_layout)
+
+
+def _add_outputs_before_if_node(model_transformer: ModelTransformer, model: ov.Model, if_node: NNCFNode) -> ov.Model:
+    """
+    Inserts extra outputs on If node inputs.
+
+    :param model_transformer: ModelTransformer instance.
+    :param model: Model instance.
+    :param if_node: If node.
+    :return: Model with extra outputs before If node.
+    """
+    transformation_layout = TransformationLayout()
+    for command in OVBackend.create_output_insertion_commands(model, if_node):
+        transformation_layout.register(command)
+    return model_transformer.transform(transformation_layout)
+
+
+def apply_algorithm_if_bodies(
+    algorithm: Algorithm,
+    parent_model: ov.Model,
+    parent_graph: NNCFGraph,
+    parent_dataset: Dataset,
+    subset_size: int,
+    parent_statistic_points: Optional[StatisticPointsContainer] = None,
+) -> ov.Model:
+    """
+
+    :param parent_model:
+    :param parent_graph:
+    :param parent_dataset:
+    :param parent_statistic_points:
+    :param parent_model_cnt:
+    :return:
+    """
+    nncf_logger.info(f"Quantize a new submodel")
```

Comment:
IMO, such logging is unnecessary for the user since the number of the `if` blocks is not limited and thus it might not look good in case of the huge number of these blocks. Especially with the logging like `The length of dataset for ..`.
I suggest moving it into the `debug` section.

Reply:
Updated. Removed 
`nncf_logger.info(f"Quantize a new submodel")`.
Also, put into the debug section the logging of dataset length.